packages/valory/skills/score_write_abci/rounds.py

```diff
@@ -87,6 +88,53 @@ def most_voted_randomness_round(self) -> int:  # pragma: no cover
         round_ = self.db.get_strict("most_voted_randomness_round")
         return cast(int, round_)
 
+    @property
+    def latest_tweet_id(self) -> int:
+        """Get the latest tracked tweet id."""
+        return cast(int, self.db.get("latest_tweet_id", 0))  # pragma: no cover
+
+
+class StartupScoreReadRound(CollectSameUntilThresholdRound):
+    """StartupScoreReadRound"""
+
+    payload_class = StartupScoreReadPayload
+    synchronized_data_class = SynchronizedData
+
+    ERROR_PAYLOAD = "error"
+    SUCCCESS_PAYLOAD = "success"
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.threshold_reached:
+            if self.most_voted_payload == self.ERROR_PAYLOAD:
+                return self.synchronized_data, Event.API_ERROR
+
+            payload = json.loads(self.most_voted_payload)
+
+            synchronized_data = self.synchronized_data.update(
+                synchronized_data_class=SynchronizedData,
+                **{
+                    get_name(SynchronizedData.user_to_total_points): payload[
+                        "user_to_total_points"
+                    ],
+                    get_name(SynchronizedData.latest_tweet_id): payload[
+                        "latest_tweet_id"
+                    ],
+                }
+            )
+
+            return synchronized_data, Event.DONE
+
+        if not self.is_majority_possible(
+            self.collection, self.synchronized_data.nb_participants
+        ):
+            return self.synchronized_data, Event.NO_MAJORITY
+        return None
```

Comment:
Even if I use mutliple fields here, I need to handle the error payload so I have to override `end_block` right?

Reply:
@Adamantios  do new payloads support passing dicts or do I need to keep stringifying them?

Reply:
> Even if I use multiple fields here, I need to handle the error payload so I have to override `end_block` right?

ahh yes, that's a pity.

> Do new payloads support passing dicts or do I need to keep stringifying them

they should :)

Reply:
When I use a dict in the payload I get in the round test:
`TypeError: unhashable type: 'dict'`
I think I need to keep the serialized version.

Reply:
Ahh ok, sorry for the frustration :disappointed:

Reply:
It might be my fault. I think let's merge this and I'll open a ticket to investigate this as both contribution and iekit need to update their payloads to use this functionality. Not only this new payload but all of them.

Reply:
Tickets opened already.